config/config.go

```diff
@@ -34,17 +34,18 @@ type ForwardTo struct {
 
 // ListenFrom configures how we listen for datapoints to forward
 type ListenFrom struct {
-	Type                       string
-	ListenAddr                 *string
-	MetricDeconstructor        *string
-	MetricDeconstructorOptions *string
-	Timeout                    *string
-	Name                       *string
-	ListenPath                 *string
-	JSONEngine                 *string
-	Encrypted                  *bool
-	TimeoutDuration            *time.Duration `json:"-"`
-	ServerAcceptDeadline       *time.Duration `json:"-"`
+	Type                           string
+	ListenAddr                     *string
+	MetricDeconstructor            *string
+	MetricDeconstructorOptions     *string
+	MetricDeconstructorOptionsJSON map[string]interface{}
```

Comment:
We parse it here as map[string]interface{} then encode and redecode to the proper structs when instantiating the particular deconstructor that is configured by json. So we get the validation for each specific deconstructor there, but not tied to a shared format for every future deconstructor that may want to be configured via json.